pilot/lib/pilot/build_manager.rb

```diff
@@ -171,6 +171,25 @@ def should_update_app_test_information?(options)
       options[:beta_app_description].to_s.length > 0 || options[:beta_app_feedback_email].to_s.length > 0
     end
 
+    # If itc_provider was explicitly specified, use it.
+    # If there are multiple teams, infer the provider from the selected team name.
+    # If there are fewer than two teams, don't infer the provider.
+    def transporter_for_selected_team(options)
+      generic_transporter = FastlaneCore::ItunesTransporter.new(options[:username], nil, false, options[:itc_provider])
+      return generic_transporter unless options[:itc_provider].nil? && @session.teams.count > 1
```

Comment:
Left a note in `pilot/manager.rb` about maybe removing the `@session` and `@selected_team_id` and doing something like this at the beginning of this method to use...
```rb
client = Spaceship::Tunes.client
team_id = client.team_id
```